### DISCUSSION TITLE
MAX31855

### DISCUSSION THREAD

**Discussion by PepeTheFroggie** [arendst/Tasmota#11135] [Q&A] [answered]
Something went wrong with the MAX31855 ... code is broken. Internal temp works while probe temp is wrong.

![max31855](https://user-images.githubusercontent.com/37778140/109294384-62b7b480-782d-11eb-91c0-ca8fdcac057b.jpg)

While the MAX6675 works perfectly.
The MAX31855 and thermocouple work win an arduino uno, not a hardware issue.

I propose to move the MAX6675 option into my_user_config instead of the cryptic
#define USE_MAX31855 and then "setoption94 1" the MAX6675 needs now.

You ask "why you use this thing"?
Exhaust temperature of my oil burner for heating the house. The DS18b20 only measures up to 127 deg C.

**Comment by PepeTheFroggie** [comment, marked as answer]
Foget it!
This could be a problem with a noisy power supply.
https://community.home-assistant.io/t/thermocouple-temperature-readings-completely-off-when-led-strip-is-turned-on/158083/9

**Comment by cpankonien** [comment]
Can I ask how you set this up? I just bought a max6675/max31855 (it's labeled with both), IO intend to use it as a sort of cooking thermometer...tell me when water is boiling, internal meat temp, etc. I can't seem to find anyone using tasmota to do this except you. Any suggestions appreciated!

**Reply by barbudor** [reply]
I also tried to use a MAX31855 + thermocouple without success
But during my test I felt that the problem was not with Tasmota talking with the MAX, but the provided ThC not to be a K-type. But I haven't been able to prove that as I don't have neither another known-to-work ThC not a Thermometer that I could test the ThC with.

This being said, configuration of Tasmota and connection was pretty straight forward.
You should be able to identify the chip by reading the label on the chip itself, including which variant it is (there are variant dedicated for each type of Thermocouple).
You have to assign 3 GPIOs to MAX31855_CS, _CLK and _DO and connect them to the appropriate pins of the module.

**Reply by techman83** [reply]
If you have a digital multi-meter, it likely have a temp probe input that is K-type. If not, you can just drop your probes across it and measure for the resistance change when you hold the probe between your fingertips. The one I have is a touch under 15 ohms at ~20°C

**Comment by cpankonien** [comment]
Yes, thanks for the response, the max31855 is indeed marked as such so that should be easy enough, figuring out which GPIOs do what will take some research (I already see one marked "clk" so that one should be easy enough), and I recall seeing recently a fairly comprehensive description of nodemcu pinouts so I'll start there. I'll be sure to post results (I've made sure mine is a K-type thermocouple).

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by barbudor** [reply]
Almost ay NodeMCU GPIO will be good for the job. There's nothing to identify here. Just choose some GPIOs and map them in Tasmota configuration to the pin name I have listed
And connect to the pin of the same name on your MAX module.

**Comment by cpankonien** [comment]
Thanks for that, I tried it and it worked, so there's one uncertainty removed! Now to figure out how to wire it up to the nodemcu, that's gonna take some research!

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by cpankonien** [comment]
First, thanks for the help! So I have the nodemcu configured in Tasmota modules as generic, and I see all the GPIOs, but none of them seem to correlate to the cs, clk or do pins on the max. Am I missing something obvious? (I have a bunch of other nodemcus configured as temp sensors using am2301 in dropdown (d2/gpio4 to data pin on am2301, ground to ground, pos to +3.3v), I assume we're going to do something similar here?) Again, thanks for the help.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by barbudor** [reply]
Which Tasmota variant are you using ?
As per the [BUILD list](https://github.com/arendst/Tasmota/blob/development/BUILDS.md), MAX31855 is available in `tasmota-sensors.bin(.gz)`

**Comment by cpankonien** [comment]
Oh, well that would explain it, I'm using tasmota.bin, let me download that and give it go!

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by cpankonien** [comment]
It works now! Apparently I need a 10nF capacitor between the pos and neg on the thermocouple. According to reviews on Amazon, it's pretty much useless with out it, so they are on order. I'm getting temps but they are all over the place, so hopefully the capacitor will fix that. Now that I know what it will look like I can start working on a 3d print of a suitable case, although I suspect I may have to take more than one shot at that, depending on how I end up using it (meat thermometer, smoker thermometer, when is the water boiling, etc.) Thanks so much for your help...not sure I'd have ever thought to try tasmota-sensors.bin, but it will be in my thought processes in the future!

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by degreal** [comment]
What is the easiest tasmota device you can connect MAX31855/MAX6675 to? I tried Sonoff Basic R3. I have no idea which GPIO pins are usable. I'm reading "null" degrees with wiring:
GPIO 9 (CS) -> CS
GPIO 10 (CLK) -> SCK
GPIO 16 (DO) -> SO

**Reply by Jason2866** [reply]
Easier is a NodeMcu. A sonoff basic is a bad choice. It is to complicate and it is dangerous since AC.
